### PR TITLE
GitHub apps pipeline support

### PR DESCRIPTION
**Pull request by v-amolpatil**
Required items, please complete
   
   Change(s):
   - Github Apps for PR without using personal token

   Reason for Change(s):
   - Github Apps for PR without using personal token

   Version Updated:
   - NA

   Testing Completed:
   - In Progress

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

image

image

image

image

### CODE REVIEW

File: .github/workflows/slash-command-dispatch.yaml [hunk 1]
```diff
@@ -24,10 +30,19 @@ jobs:
     needs: [pullRequestStatus, checkAutomatedPR]
     if: ${{ needs.pullRequestStatus.outputs.isPullRequestMerged == 'False' && needs.checkAutomatedPR.outputs.isAutomatedPR == 'False' && !github.event.pull_request.head.repo.fork}}
     steps:
+      - name: Generate a token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
```
Comment (Tichandr) [inline .github/workflows/slash-command-dispatch.yaml:35]: looks like this is using third party action. Is there anything official action from github available for token generation?

File: .github/workflows/package-command.yaml [hunk 2]
```diff
@@ -2,136 +2,102 @@ name: Package Command
...
-
-            if ($solutionName -eq '' -or $null -eq $solutionName)
-            {
-              Write-Output "::warning:: Package-Creation : Skipping as Solution name is blank."
-              Send-AppInsightsTraceTelemetry -InstrumentationKey $instrumentationKey -Message "Execution for create-package started and received blank solutionName for Job Run Id : $runId" -Severity Information -CustomProperties $customProperties
-            }
-            else
-            {
-              Send-AppInsightsTraceTelemetry -InstrumentationKey $instrumentationKey -Message "Execution for create-package started for Solution Name : $solutionName, Job Run Id : $runId" -Severity Information -CustomProperties $customProperties
+          token: ${{ env.GeneratedToken}}
```
Comment (Tichandr) [inline .github/workflows/package-command.yaml:40]: what is this token assignment?

File: .github/workflows/package-command.yaml [hunk 3]
```diff
@@ -2,136 +2,102 @@ name: Package Command
...
-            exit 1
+            Write-Host "Skipping packaging as it is an automated pr!"
+            $isAutomatedPR = $true
           }
+          Write-Output "isAutomatedPR=$isAutomatedPR" >> $env:GITHUB_OUTPUT
+          Write-Host "Is this Pull Request autogenerated $isAutomatedPR"
+
+      - id: validateAndCreatePackage
+        name: validate-create-package
+        if: ${{ success() && steps.checkAutomatedPR.outputs.isAutomatedPR == 'False' }}
```
Comment (Tichandr) [inline .github/workflows/package-command.yaml:61]: check the false conditioning, is it case sensitive in above step?

File: .github/workflows/package-command.yaml [hunk 4]
```diff
@@ -2,136 +2,102 @@ name: Package Command
...
-            }
-          }
-          catch
+      - id: checkAutomatedPR
+        name: check-Automated-PR
+        env:
+          BranchName: ${{ github.event.pull_request.head.ref || github.event.client_payload.pull_request.head.ref || github.event.client_payload.pullRequestBranchName }}
+        shell: pwsh
+        run: |
+          $pullRequestBranchName = "$env:BranchName"
```
Comment (Tichandr) [inline .github/workflows/package-command.yaml:48]: will this give you the branch name or it must be something like this ${{ env.BranchName}}. plz check

File: .github/workflows/package-command.yaml (hunk 3 above)
Comment (v-amolpatil) [reply .github/workflows/package-command.yaml:61]: verified and its giving the correct branch name

File: .github/workflows/package-command.yaml (hunk 2 above)
Comment (v-amolpatil) [reply .github/workflows/package-command.yaml:40]: this is a github app token that we are using.

File: .github/workflows/package-command.yaml (hunk 3 above)
Comment (v-amolpatil) [reply .github/workflows/package-command.yaml:61]: image

**Review by Tichandr — APPROVED** [review]
Looks Good. Approving.

**Pull request merged**